src/app/modules/main/chat_section/controller.nim

```diff
@@ -468,10 +468,12 @@ proc getStatusForContactWithId*(self: Controller, publicKey: string): StatusUpda
   return self.contactService.getStatusForContactWithId(publicKey)
 
 proc acceptContactRequest*(self: Controller, publicKey: string, contactRequestId: string) =
-  self.contactService.acceptContactRequest(publicKey, contactRequestId)
+  let response = self.contactService.acceptContactRequest(publicKey, contactRequestId)
+  discard self.chatService.processMessageUpdateAfterSend(response)
```

Comment:
I think it would be cleaner if the calls to `processMessageUpdateAfterSend` were made from services only.

I know there would be a cyclic dependency between the chat and contact service, but maybe we could do an event instead.

In the chat service, you could add an event like `self.events.on(PROCESS_MESSAGE_UPDATE, ...)` where the Arg is the RPC response that calls `processMessageUpdateAfterSend`. 

IMO that's cleaner than having to pass the response back to the controller.

Reply:
Fixed!